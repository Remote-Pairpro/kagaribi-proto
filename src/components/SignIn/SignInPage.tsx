import React from "react";
import { withRouter } from "react-router-dom";
import { SignUpLink } from "../SignUp/SignUp";
import SignInForm, { ISignInFormProps } from "./SignInForm";

const SignInPage = (signInFormProps: ISignInFormProps) => {
    return (
        <div>
            <h1>SignIn</h1>
            <SignInForm history={signInFormProps.history} />
            <SignUpLink />
        </div>
    );
};

export default withRouter(SignInPage);
