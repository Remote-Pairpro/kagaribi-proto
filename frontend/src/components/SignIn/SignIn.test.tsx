import { shallow } from "enzyme";
import { createMemoryHistory } from "history";
import React from "react";
import SignInFrom from "./SignInForm";
import SignInPage from "./SignInPage";

describe("<SignInForm />", () => {
    it("should be rendered correctly", () => {
        const history = createMemoryHistory();
        const tree = shallow(<SignInFrom history={history} />, {
            disableLifecycleMethods: true,
        });
        expect(tree).toMatchSnapshot();
    });
});

describe("<SignInPage />", () => {
    it("should be rendered correctly", () => {
        const tree = shallow(<SignInPage />);
        expect(tree).toMatchSnapshot();
    });
});
