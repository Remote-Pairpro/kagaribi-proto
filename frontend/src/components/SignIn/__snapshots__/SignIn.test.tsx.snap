// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInForm /> should be rendered correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <TextField
    autoComplete="on"
    id="email-address"
    label="Email"
    onChange={[Function]}
    placeholder="Email Address"
    required={false}
    select={false}
    type="text"
    value=""
    variant="standard"
  />
    
  <TextField
    autoComplete="on"
    id="password"
    label="Password"
    onChange={[Function]}
    placeholder="Passwprd"
    required={false}
    select={false}
    type="password"
    value=""
    variant="standard"
  />
  <WithStyles(Button)
    disabled={true}
    type="submit"
    variant="outlined"
  >
    Sign In
  </WithStyles(Button)>
</form>
`;

exports[`<SignInPage /> should be rendered correctly 1`] = `
<Route>
  <Component />
</Route>
`;
